jax/_src/util.py

```diff
@@ -185,6 +187,18 @@ def cache(max_size=4096):
 
 memoize = functools.lru_cache(maxsize=None)
 
+def memoize_x64(f):
+  """Memoization which returns different results depending on the X64 flag."""
+  f_memoized_x32 = memoize(f)
+  f_memoized_x64 = memoize(f)
```

Comment:
Instead of having separate caches, another way to do this would be to add dummy args for the things we want to be sensitive to, maybe like

```
def memoize(f):
  @functools.lru_cache(maxsize=None)
  def f_memoized(_, *args, **kwargs):
    return f(*args, **kwargs)

  @functools.wraps(f)
  def wrapper(*args, **kwargs):
    return f_memoized(FLAGS.jax_enable_x64, *args, **kwargs)

  return wrapper   
```

This might work nicely for `cache`, where we want to control the max size. It also might scale better to more context, e.g. if we have multiple flags of context we want to include.

Reply:
Done - please take a look!